atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/SweepStateCoordinator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.asts;
+
+import com.palantir.atlasdb.sweep.queue.ShardAndStrategy;
+import com.palantir.logsafe.Safe;
+import java.util.function.Consumer;
+import org.immutables.value.Value;
+
+public interface SweepStateCoordinator {
+    State tryRunTaskWithSweep(Consumer<SweepableBucket> task);
+
+    enum State {
```

Comment:
Hmm, `SweepOutcome` or similar - also I think we said `SWEPT` might be a bit confusing since it doesn't guarantee that the task swept everything in the bucket?